File: ThreadsApp/ThreadsApp/View/Chat/Cells/Text/TextCell.m
//
//  TextCell.m
//  ThreadsApp
//
//  Created by Tabriz Dzhavadov on 13/07/2017.
//  Copyright © 2017 Sequenia. All rights reserved.
//

#import "TextCell.h"

@interface TextCell()

@property (weak, nonatomic) IBOutlet UITextField *textField;

@end

@implementation TextCell

#pragma mark - Initialize

- (void)awakeFromNib {
    [super awakeFromNib];
    self.selectionStyle = UITableViewCellSelectionStyleNone;
}

#pragma mark - Configuring

- (void) construct {
    self.textField.placeholder = [CellTypeHelper textByType: self.type];
    [self.textField addTarget: self
                       action: @selector(textFieldChanged:)
             forControlEvents: UIControlEventEditingChanged];
}

- (void) setText:(NSString *)text {
    [self.textField setText: text];
    [self textFieldChanged: self.textField];
}

- (NSString *) text {
    return self.textField.text;
}

- (void) textFieldChanged: (UITextField *) textField {
    if (self.delegate && [self.delegate respondsToSelector: @selector(textCell:textChanged:)]) {
        [self.delegate textCell: self textChanged: textField.text];
    }
}

@end

File: ThreadsApp/ThreadsAppSwift/View/Chat/Cells/TextCell.swift
//
//  TextCell.swift
//  THRAppSwift-Polarbear
//
//  Created by Brooma Service on 16/05/2019.
//  Copyright © 2019 Brooma Service. All rights reserved.
//

import Foundation
import UIKit

protocol TextCellDelegate: class {
    func textChanged(_ text: String?, inCell: TextCell)
}

class TextCell : UITableViewCell {
    
    weak var delegate : TextCellDelegate?
    
    var type: CellType!
    
    @IBOutlet weak var textField : UITextField!
    
    //MARK: - Initialize
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.selectionStyle = .none
    }
    
    //MARK: - Configuring
    
    func construct() {
        self.textField.placeholder = self.type.rawValue
    }
    
    func setText(_ text: String) {
        self.textField.text = text
        self.textFieldChanged(self.textField!)
    }
    
    func getText() -> String? {
        return self.textField?.text
    }
    
    @IBAction func textFieldChanged(_ sender: Any) {
        delegate?.textChanged(textField.text, inCell: self)
    }
}
